src/storage/src/hummock/compactor/compaction_utils.rs

```diff
@@ -219,7 +219,7 @@ pub async fn generate_splits(compact_task: &mut CompactTask, context: Arc<Compac
             );
         }
         // sort by key, as for every data block has the same size;
-        indexes.sort_by(|a, b| KeyComparator::compare_encoded_full_key(a.1.as_ref(), b.1.as_ref()));
+        indexes.sort_by(|a, b| FullKey::decode(a.1.as_ref()).cmp(&FullKey::decode(b.1.as_ref())));
```

Comment:
I think this change is unneccessary since we already have the encoded full key for free so using `compare_encoded_full_key` is fine.

Reply:
Do you think it's ok for the changes in sst_iter seek?

Reply:
At first I tend to remove `compare_encoded_full_key`, and compare fullkey struct every where, but in some places there are already two encoded keys. Or just unify the `smallest_key`, `largest_key`, etc. to fullkey struct.

Reply:
> I think this change is unneccessary since we already have the encoded full key for free so using compare_encoded_full_key is fine.

We may be able to save the encode in L215 though.

Reply:
Let's remain key_range unchanged, and just refactor sst iter in this PR.